hedera-node/src/main/java/com/hedera/services/txns/schedule/ScheduleSignTransitionLogic.java

```diff
@@ -56,71 +48,44 @@
 	private final Function<TransactionBody, ResponseCodeEnum> SEMANTIC_CHECK = this::validate;
 
 	private final InHandleActivationHelper activationHelper;
-	private final ScheduleStore store;
-	private final ScheduleModelStore modelStore;
+	private final ScheduleModelStore store;
 	private final TransactionContext txnCtx;
 	SigMapScheduleClassifier classifier = new SigMapScheduleClassifier();
 	SignatoryUtils.ScheduledSigningsWitness replSigningsWitness = SignatoryUtils::witnessScoped;
-	private ScheduleExecutor executor;
 
 	@Inject
 	public ScheduleSignTransitionLogic(
-			ScheduleStore store,
-			ScheduleModelStore modelStore,
+			ScheduleModelStore store,
 			TransactionContext txnCtx,
-			InHandleActivationHelper activationHelper,
-			ScheduleExecutor executor
+			InHandleActivationHelper activationHelper
 	) {
-		this.store = store;
 		this.txnCtx = txnCtx;
-		this.executor = executor;
 		this.activationHelper = activationHelper;
-		this.modelStore = modelStore;
+		this.store = store;
 	}
 
 	@Override
 	public void doStateTransition() {
-		try {
-			var accessor = txnCtx.accessor();
-			transitionFor(accessor.getSigMap(), accessor.getTxn().getScheduleSign());
-		} catch (Exception e) {
-			log.warn("Unhandled error while processing :: {}!", txnCtx.accessor().getSignedTxnWrapper(), e);
-			txnCtx.setStatus(FAIL_INVALID);
-		}
-	}
+		var accessor = txnCtx.accessor();
+		var scheduleID = accessor.getTxn().getScheduleSign().getScheduleID();
+		var schedule = store.loadSchedule(Id.fromGrpcSchedule(scheduleID));
 
-	private void transitionFor(
-			SignatureMap sigMap,
-			ScheduleSignTransactionBody op
-	) throws InvalidProtocolBufferException {
-		var scheduleId = op.getScheduleID();
-		var origSchedule = store.get(scheduleId);
-		if (origSchedule.isExecuted()) {
-			txnCtx.setStatus(SCHEDULE_ALREADY_EXECUTED);
-			return;
-		}
-		if (origSchedule.isDeleted()) {
-			txnCtx.setStatus(SCHEDULE_ALREADY_DELETED);
-			return;
-		}
-		var schedule = new Schedule(Id.fromGrpcSchedule(scheduleId)); // NOTE todo
-		ScheduleConversion.merkleToModel(origSchedule, schedule);
 		var validScheduleKeys = classifier.validScheduleKeys(
 				List.of(txnCtx.activePayerKey()),
-				sigMap,
+				accessor.getSigMap(),
 				activationHelper.currentSigsFn(),
 				activationHelper::visitScheduledCryptoSigs);
-		// NOTE todo
 		var signingOutcome = replSigningsWitness.observeInScope(schedule, validScheduleKeys, activationHelper);
 
 		var outcome = signingOutcome.getLeft();
 		if (outcome == OK) {
-			var updatedSchedule = store.get(scheduleId);
-			txnCtx.setScheduledTxnId(updatedSchedule.scheduledTransactionId());
+			txnCtx.setScheduledTxnId(schedule.scheduledTransactionId());
 			if (signingOutcome.getRight()) {
-				outcome = executor.processExecution(scheduleId, store, txnCtx);
+				schedule.execute(txnCtx, store::resetPendingCreation);
 			}
 		}
+
```

Comment:
The previous execution outcome was coming from the `executor.processExecution` method. We must think of a solution to either:
- set the status in `schedule.execute` method
- assume it's successful ( it throws `InvalidTransactionException` on fail )

Reply:
Let's also check `SignatoryUtils` for something buggy